src/main/java/org/spongepowered/api/item/enchantment/Enchantment.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.enchantment;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.persistence.DataBuilder;
+import org.spongepowered.api.item.inventory.ItemStack;
+
+/**
+ * Represents an {@link EnchantmentType} on an {@link ItemStack} that is paired
+ * with a level.
+ *
+ * <p>The contract of enchantments is that their level will always be larger
+ * than 0 and never larger than {@link Short#MAX_VALUE}.</p>
+ */
+public interface Enchantment extends DataSerializable {
+
+    /**
+     * Creates a new {@link Builder} to create an {@link Enchantment}.
+     *
+     * @return The new builder
+     */
+    static Builder builder() {
+        return Sponge.getRegistry().createBuilder(Enchantment.Builder.class);
+    }
+
+    /**
+     * Creates a new {@link Enchantment} with the provided
+     * {@link EnchantmentType} and level.
+     *
+     * @param enchantmentType The enchantment type
+     * @param level The enchantment level
+     * @return The created enchantment
+     */
+    static Enchantment of(EnchantmentType enchantmentType, int level) {
+        return builder()
+                .type(enchantmentType)
+                .level(level)
+                .build();
+    }
+
+    /**
+     * Gets the {@link EnchantmentType} for this enchantment.
+     *
+     * @return The enchantment type of this enchantment
+     */
+    EnchantmentType getType();
+
+    /**
+     * Gets the level of this enchantment.
+     *
+     * @return The level of this enchantment
+     */
+    int getLevel();
+
+    /**
+     * Represents a builder interface which can be used
+     * to create a {@link Enchantment}.
+     */
+    interface Builder extends DataBuilder<Enchantment> {
+
+        /**
+         * Sets the {@link EnchantmentType} for this enchantment.
+         *
+         * @param enchantmentType The desired enchantment type
+         * @return The modified builder, for chaining
+         */
+        Builder type(EnchantmentType enchantmentType);
+
+        /**
+         * Sets the level for this enchantment.
+         *
+         * @param level The desired level
+         * @return The modified builder for chaining
+         * @throws IllegalArgumentException If the level is below 1
```

Comment:
This contradicts the argument in the java doc for the class of `Enchantment` [here](https://github.com/SpongePowered/SpongeAPI/pull/1687/files#diff-5e7629a1c106a95db1ab90b7621fc617R37)

Reply:
And I'm somewhat concerned about mod compatibility here.

Reply:
There I said their level will always be larger than 0, which equates to the same thing as never below 1, but I can clean up the wording and add the max as well. 

How do you propose I handle the concern? Should we simply allow them to use the full range of `Short.MIN_VALUE` to `Short.MAX_VALUE`, specifying that unintentional issues may arise if you go outside of `EnchantmentType#getMinumumLevel` and `EnchantmentType#getMaxmimumLevel`?

Reply:
> Should we simply allow them to use the full range of Short.MIN_VALUE to Short.MAX_VALUE, specifying that unintentional issues may arise if you go outside of EnchantmentType#getMinumumLevel and EnchantmentType#getMaxmimumLevel?

I am going to have to say "yes" because we cannot 100% control the possibilities of enchantments. If developers choose to make unworkable enchantments, that's more on them.

Reply:
I will make this change and the necessary changes to the docs later tomorrow.